lib/CL/clCreateKernel.c

```diff
@@ -76,8 +78,10 @@ POname(clCreateKernel)(cl_program program,
       if (program->binaries[device_i] &&
           pocl_cache_device_cachedir_exists(program, device_i))
         {
+#ifdef OCS_AVAILABLE
           pocl_llvm_get_kernel_metadata(program, kernel, device_i,
                                         kernel_name, &errcode);
+#endif
```

Comment:
@pjaaskel for fetching metadata from the binary dump there's ```pocl_binary_get_kernel_metadata(...) ```

pocl_llvm_get_kernel_metadata() is only for program.bc LLVM bitcode.